import java.util.function.Function;

public class TestMethodRef {

	public static void main(String[] args) {
		Function<String, Integer> f = Integer::new;
	}

}

This is accepted by javac(of course, 8) but throw an error in my onw built BETA_JAVA8. From the semantics of lambda, it should be ok. Note, BETA_JAVA8 is a method ref, but not a method invocation.

This is so common, but we forget to implement it? Or, may I have a wrong build for BETA_JAVA8?

Anywhere, team, good work in these months! 

thanks, 
Jin

import java.util.function.Function;

public class TestMethodRef {

	public static void main(String[] args) {
		Function<String, Integer> f = Integer::new;
	}

}

This is accepted by javac(of course, 8) but throw an error in my onw built BETA_JAVA8. From the semantics of lambda, it should be ok. Note, BETA_JAVA8 is a method ref, but not a method invocation.

This is so common, but we forget to implement it? Or, may I have a wrong build for BETA_JAVA8?

Anywhere, team, good work in these months! 

thanks, 
Jin
Thanks for the report, I'll take a look.
Anirban, please take a look. TIA.
Created attachment 233961
Patch for the fix

Patch for the fix
Patch looks good. Jay I am having some trouble releasing the patch. Can you
please release ? TIA.
(In reply to comment #4)
> Patch looks good. Jay I am having some trouble releasing the patch. Can you
> please release ? TIA.

Released via:
http://git.eclipse.org/c/jdt/eclipse.jdt.core.git/commit/?h=BETA_JAVA8&id=010dc9e8d593c0ba9b4912ff35db0ecfac142eab

Anirban, I didn't run the tests, I am assuming you already did.
(In reply to comment #5)
> (In reply to comment #4)
> > Patch looks good. Jay I am having some trouble releasing the patch. Can you
> > please release ? TIA.
> 
> Released via:
> http://git.eclipse.org/c/jdt/eclipse.jdt.core.git/commit/
> ?h=BETA_JAVA8&id=010dc9e8d593c0ba9b4912ff35db0ecfac142eab
> 
> Anirban, I didn't run the tests, I am assuming you already did.

Thanks Jay; I ran the tests


[1.8][compiler] Bogus error about unhandled exceptions for unchecked exceptions thrown by method reference.



Fix for Bug 411273 -[1.8][compiler] Bogus error about unhandled exceptions for unchecked exceptions thrown by method reference. 